### PR TITLE
Development for 3.2.x

### PR DESCRIPTION
**Pull request by VSEphpbb**
Not able to be backwards compatible with 3.1. So this is now 3.0.0-dev.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
     "type": "phpbb-extension",
     "description": "phpBB Ideas extension",
     "homepage": "https://www.phpbb.com",
-    "version": "2.0.0-dev",
+    "version": "3.0.0-dev",
```
Comment (michaelcullum) [inline composer.json]: 2.1. Let's not get ahead of ourselves.

Comment (VSEphpbb) [inline composer.json]: Well, semver ;) This isn't BC with 2.0.0

Comment (michaelcullum) [inline composer.json]: hm.

File: language/en/common.php [hunk 2]
```diff
@@ -94,7 +94,7 @@
 	'VIEW_ALL'				=> 'View All',
 	'VIEW_IDEA'				=> 'View Idea',
 	'VIEW_IDEA_AT'          => 'View idea at: %s', // Warning: submitted to db
-	'VIEW_VOTES'            => '%s points. Click to view votes',
+	'VIEW_VOTES'            => '%s point(s). Click to view votes',
```
Comment (rxu) [inline language/en/common.php]: Can be converted to plural form.

Comment (VSEphpbb) [inline language/en/common.php]: I already tried. This key is swapped back and forth by javascript when voting and couldn't get it to work out.

File: styles/prosilver/template/idea_body.html [hunk 3]
```diff
@@ -50,86 +54,86 @@
...
-			<a href="{U_EDIT_RFC}" id="rfcedit" data-l-add="{L_ADD}" data-l-edit="{L_EDIT}"><!-- IF IDEA_RFC -->{L_EDIT}<!-- ELSE -->{L_ADD}<!-- ENDIF --></a>
-			<input type="text" id="rfceditinput" value="{IDEA_RFC}" data-l-err="{L_ERROR}" data-l-msg="{L_RFC_ERROR}" />
+						<!-- BEGIN votes_down --><!-- IF votes_down.S_VOTED --><li><a href="{U_REMOVE_VOTE}" class="removevote" data-l-err="{L_ERROR}" data-l-msg="{L_VOTE_ERROR}{L_COLON}">{L_REMOVE_VOTE}</a></li><!-- ENDIF --><!-- END votes_down -->
+						<!-- ENDIF -->
+					</ul>
+				</div>
+			</dd>
+			<dt style="width: 15%">{L_STATUS}:</dt>
+			<dd>
+				<a href="{IDEA_STATUS_LINK}">{IDEA_STATUS}</a>
```
Comment (rxu) [inline styles/prosilver/template/idea_body.html]: Placeholders in href should probably all have the ```U_``` prefix for consistency.

File: styles/prosilver/template/idea_body.html [hunk 4]
```diff
@@ -50,86 +54,86 @@
...
+
+				<a id="ticketlink" <!-- IF IDEA_TICKET -->href="https://tracker.phpbb.com/browse/PHPBB3-{IDEA_TICKET}">PHPBB3-{IDEA_TICKET}<!-- ELSE -->style="display:none"><!-- ENDIF --></a>
+
+				<!-- IF CAN_EDIT -->
+				<a href="{U_EDIT_TICKET}" id="ticketedit" data-l-add="{L_ADD}" data-l-edit="{L_EDIT}"><!-- IF IDEA_TICKET -->{L_EDIT}<!-- ELSE -->{L_ADD}<!-- ENDIF --></a>
+				<input type="text" id="ticketeditinput"<!-- IF IDEA_TICKET --> value="PHPBB3-{IDEA_TICKET}"<!-- ENDIF --> placeholder="PHPBB3-#####" data-l-err="{L_ERROR}" data-l-msg="{L_TICKET_ERROR}" />
+				<!-- ENDIF -->
+			</dd>
+			<!-- ENDIF -->
+			<!-- IF IDEA_DUPLICATE or IS_MOD -->
```
Comment (rxu) [inline styles/prosilver/template/idea_body.html]: We use ```S_``` prefix for switches, don't know how much does it matter here and in other places.

Comment (VSEphpbb) [inline styles/prosilver/template/idea_body.html]: yep.good catches

File: composer.json (hunk 1 above)
Comment (Elsensee) [inline composer.json]: Where were you when the decision was made to have phpBB 3.1?

Comment (callumacrae) [inline composer.json]: If you're using semver, you're allowed to break compatibility in unstable versions as much as you like. `2.0.0-dev2` can be a complete rewrite of `2.0.0-dev`. That said, if there are a lot of users, probably not a great idea. (there aren't, though)

**Pull request merged**